gammapy/stats/tests/test_variability.py

```diff
@@ -161,31 +161,21 @@ def temp(x, norm, index):
     params = {"norm": 1.5, "index": 3}
 
     time_series3, time_axis3 = TimmerKonig_lightcurve_simulator(
-        temp, 15, 1 * u.h, power_spectrum_params=params
+        temp, 15, 1 * u.h, nchunks=100, power_spectrum_params=params
     )
 
     time_series4, time_axis4 = TimmerKonig_lightcurve_simulator(
-        lambda x: x ** (-3), 20, 1 * u.s, leakage_protection=100.0
+        lambda x: x ** (-3), 200, 1 * u.s, mean=2.0, std=0.1
     )
 
     assert len(time_series) == 20
     assert isinstance(time_axis, u.Quantity)
     assert time_axis.unit == u.s
     assert len(time_series2) == 21
     assert len(time_series3) == 15
-    assert len(time_series4) == 20
-
-
-def test_tk_badleakage():
-    with pytest.raises(ValueError):
-        TimmerKonig_lightcurve_simulator(
-            lambda x: x ** (-3), 20, 1 * u.s, leakage_protection=0.1
-        )
-
-    with pytest.raises(Warning):
-        TimmerKonig_lightcurve_simulator(
-            lambda x: x ** (-3), 20, 1 * u.s, leakage_protection=1.001
-        )
+    assert len(time_series4) == 200
```

Comment:
Shouldn't the LC be shrunked to the input size?

Reply:
In this case input size is 200 so it is the correct length

Reply:
My bad :)
Then it would be worth testing that changing nchunks always return the correct size.